vars/buildInfoYaml.groovy

```diff
@@ -0,0 +1,62 @@
+def call(Map args = [:]) {
+
+    try {
+        unstash "buildInfo_yml"
+    } catch(Exception ex) {
+        echo("No buildInfo.yml exists in stashed. Starting initialize the buildInfo yaml file.")
+    }
+
+    def inputManifest = args.inputManifest ?: "buildInfo.yml"
+    def sourceyml = readYaml(file: inputManifest)
+    def outputyml = args.outputyml ?: "buildInfo.yml"
+    def components = args.componentName
+    def componentsList = []
+    def status = args.status
+    echo("The status is $status")
+    echo("Components is $components")
+    if (!components.isEmpty()) {
+        echo ("Components parameter is not null")
+        for (component in components.split(" ")) {
+            componentsList.add(component.trim())
+        }
+    } else {
+        echo ("Components parameter is null")
+        sourceyml.components.each { component ->
+            componentsList.add(component.name)
+        }
+    }
+    echo (componentsList.toString())
+
+    if (args.stage == "INITIALIZE_STAGE") {
```

Comment:
I still find this big state machine quite problematic. I won't block this PR for it, but I think the biggest bug here is that your workflow will silently produce an incorrect YML if you forget to call the function with `INITIALIZE_STAGE`.

I would extract the initialization into its own function in which you do not `unstash` an existing build info and remove the `catch` in unstashing it - that should never happen unless there's a bug in the workflow.